Update docs related to React 18 (#35952)
vercel/next.js: docs/advanced-features/react-18/streaming.md

# Streaming SSR

> React 18 includes architectural improvements to React server-side rendering (SSR) performance. This means you can use `Suspense` in your React components in streaming SSR mode and React will render them on the server and send them through HTTP streams.

Reviewer: ```suggestion
React 18 includes architectural improvements to React server-side rendering (SSR) performance. This means you can use `Suspense` in your React components in streaming SSR mode and React will render content on the server and send updates through HTTP streams.
```

> It's worth noting that the experimental React Server Components feature, is based on streaming. You can read more about server components related streaming APIs in [`next/streaming`](/docs/api-reference/next/streaming.md). However, this guide focuses on basic React 18 streaming.

Reviewer: ```suggestion
React Server Components, an experimental feature, is based on streaming. You can read more about Server Components related streaming APIs in [`next/streaming`](/docs/api-reference/next/streaming.md). However, this guide focuses on streaming with React 18.
```

> ## Use Streaming SSR

Reviewer: ```suggestion
## Using Streaming Server-Rendering
```

> When you have Suspense in use in a SSR page, there is no extra configuration required to use streaming SSR.

Reviewer: ```suggestion
When you use Suspense in a server-rendered page, there is no extra configuration required to use streaming SSR.
```

## Enable Streaming SSR
## Use Streaming SSR

Enabling streaming SSR means React renders your components into streams and the client continues receiving updates from these streams even after the initial SSR response is sent. In other words, when any suspended components resolve down the line, they are rendered on the server and streamed to the client. With this strategy, the app can start emitting HTML even before all the data is ready, improving your app's loading performance. As an added bonus, in streaming SSR mode, the client will also use selective hydration strategy to prioritize component hydration which based on user interaction.
When you have Suspense in use in a SSR page, there is no extra configuration required to use streaming SSR.

> To enable streaming SSR, set the experimental option `runtime` to either `'nodejs'` or `'edge'`:

Reviewer 2: The user doesn't need to use the Edge runtime? What happens if they use the Node.js runtime? Do they still get streaming?

Reviewer: We should probably clarify this - you could stream using the Node.js runtime if you were self-hosting.

Author: No they don't need to "enable" streaming anymore, the rendering architecture is always enabled now. If they use the Edge runtime, they get streaming from the Edge. If they use the Node.js runtime, they will have streaming if Next.js is running in a Node.js server. And if it's running in a lambda function that doesn't support streaming, the response will be buffered and sent all at once when everything is ready.

> All SSR pages have the ability to render components into streams and the client continues receiving updates from these streams even after the initial SSR response is sent. In other words, when any suspended components resolve down the line, they are rendered on the server and streamed to the client. With this strategy, the app can start emitting HTML even before all the data is ready, improving your app's loading performance. As an added bonus, in streaming SSR mode, the client will also use selective hydration strategy to prioritize component hydration which based on user interaction.

Reviewer: ```suggestion
All SSR pages have the ability to render components into streams and the client continues receiving updates from these streams even after the initial SSR response is sent. When any suspended components resolve down the line, they are rendered on the server and streamed to the client. This means applications can start emitting HTML even _before_ all the data is ready, improving your app's loading performance.

As an added bonus, in streaming SSR mode the client will also use selective hydration to prioritize component hydration based on user interactions, further improving performance.
```

# Streaming SSR

React 18 includes architectural improvements to React server-side rendering (SSR) performance. This means you can use `Suspense` in your React components in streaming SSR mode and React will render content on the server and send updates through HTTP streams.
React Server Components, an experimental feature, is based on streaming. You can read more about Server Components related streaming APIs in [`next/streaming`](/docs/api-reference/next/streaming.md). However, this guide focuses on streaming with React 18.

## Using Streaming Server-Rendering

> When you use Suspense in a server-rendered page, there is no extra configuration required to use streaming SSR.

Reviewer: ```suggestion
When you use Suspense in a server-rendered page, there is no extra configuration required to use streaming SSR. When deployed, streaming can be utilized through infrastructure like [Edge Functions](https://vercel.com/edge) on Vercel (with the Edge Runtime) or with a Node.js server (with the Node.js runtime). AWS Lambda Functions do not currently support streaming responses.
```
